release-process.rst

```diff
@@ -28,51 +34,91 @@ Roughly:
 
 No feature addition after final x.y.0 release (or x.0.0).
 
-Backward compatibility MUST be respected within the same major release (e.g.,
-8.x.x). Binary compatibility (API or ABI) MAY be broken between two features
-releases, f.e. between 8.3 and 8.4.
+Backward Compatibility
+======================
+
+A backward compatibility (BC) break is defined as any change that prevents
+existing, valid, userland code from continuing to behave as it did in a previous
+version within the same major release.
+
+The following are not considered BC breaks:
+
+-  Adding deprecations. Code that triggers a deprecation warning continues to
+   work and is still valid. Converting deprecations into exceptions is a user
+   choice and not part of the language's default behavior.
+
+-  Adding new symbols (e.g., functions, classes, constants) into the global or
+   core extension namespace, even if they may conflict with user-defined names.
+   While these additions can cause name conflicts, they are not classified as BC
+   breaks. RFCs and contributors SHOULD make a best effort to minimize the risk
+   of conflicts when choosing new names, but SHOULD NOT pick significantly worse
+   names purely to reduce conflict risk.
+
+-  Behavior changes in undefined or undocumented edge cases MAY be allowed if
+   well justified. However, care SHOULD be taken to minimize disruption.
 
 Major Version Number
 ====================
 
 -  x.y.z to x+1.0.0
 
-   -  Bug fixes
-   -  New features
-   -  Extensions support can be ended (moved to PECL)
-   -  Backward compatibility can be broken
-   -  API compatibility can be broken (internals and userland)
-   -  ABI can be broken (internals)
+   -  It SHALL included bugfixes and new features.
+   -  Extensions support MAY be ended (moved to PECL)
+   -  Backward compatibility MAY be broken
+   -  API compatibility MAY be broken (internals and userland)
+   -  ABI MAY be broken (internals)
 
 Minor Version Number
 ====================
 
 -  x.y.z to x.y+1.z
 
-   -  Bugfixes
-   -  New features
-   -  Extensions support can be ended (moved to PECL)
-   -  Backward compatibility must be kept
-   -  API compatibility must be kept (userland)
-   -  ABI and API can be broken (internals)
-   -  Source compatibility should be kept if possible, while breakages are
+   -  It SHOULD included bugfixes and new features.
+
+   -  Extensions support MAY be ended (moved to PECL).
+
+   -  Syntax backward compatibility SHOULD be preserved - every PHP program that
+      compiles SHOULD continue to compile.
+
+   -  Backwards compatibility breaks in minor versions MUST NOT result in silent
+      behavioral differences. Instead any breaking change MUST be "obvious" when
+      executing the program.
```

Comment:
Can you clarify what this looks like in practice? This makes sense when introducing new error cases, i.e. things that now throw an exception. But it doesn't make much sense to change the behavior of non-error cases while then emitting some notice or warning, if the code isn't otherwise considered problematic. If it _is_ problematic, behavior should not change and a deprecation should be emitted instead, so it can later be removed.

Reply:
I think the point here is that it should never change for non error cases. For such cases, the deprecation would be used which is not considered as a BC break.

Reply:
I noted the error bit there. Let me know if it's ok. Maybe it should be SHOULD?

Reply:
@iluuu1994 Given I was the one who suggested this, to give an example: https://wiki.php.net/rfc/static_variable_inheritance would have been an illegal BC break with this policy.